### PR TITLE
traefik: added pod disruption budget

### PR DESCRIPTION
**Pull request by caarlos0**
**What this PR does / why we need it**:

Added a `PodDisruptionBudget` for it, so it can be Production Ready™️.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

---

cc/ @krancour

### CODE REVIEW

**Comment by timstoop** [comment]
I think this is a really nice addition, could you rebase and bump the version again, please?

**Comment by caarlos0** [comment]
@timstoop sure, done!

File: stable/traefik/README.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ The following table lists the configurable parameters of the Traefik chart and t
 | `loadBalancerSourceRanges`      | list of IP CIDRs allowed access to load balancer (if supported)      | None                                      |
 | `externalTrafficPolicy`         | Set the externalTrafficPolicy in the Service to either Cluster or Local | `Cluster`                              |
 | `replicas`                      | The number of replicas to run; __NOTE:__ Full Traefik clustering with leader election is not yet supported, which can affect any configured Let's Encrypt setup; see Clustering section | `1` |
+| `minAvailable`                  | minimum number of available pods for PodDisruptionBudget             | `1` |
```
Comment (krancour) [inline stable/traefik/README.md]: imo, the name of the option, `minAvailable`, is not adequately self-descriptive. "minAvailable" doesn't provide sufficient context. Minimum what? For what? The documentation is good, but the option name itself should provide context.

Comment (timstoop) [reply stable/traefik/README.md]: `minAvailablePods` then? Or make it something like `pdb.minAvailable`, so it's clear that this is the `minAvailable` option of the PDB?

Comment (caarlos0) [reply stable/traefik/README.md]: yeah, I'm not sure it's a good name either, I was just following the pattern: https://github.com/kubernetes/charts/search?utf8=%E2%9C%93&q=minavailable&type=

Comment (krancour) [reply stable/traefik/README.md]: Ah. I didn't realize this was an established pattern.

Let's ask someone who's more involved across a wider sampling of charts whether this is a convention that should be observed.

@prydonius or @unguiculus maybe one of you gentlemen could weigh in?

Comment (unguiculus) [reply stable/traefik/README.md]: I wouldn't say it's established and I'd like to advise for something like this, so users have the option whether they want `minAvailable` or `maxUnavailable`.

https://github.com/kubernetes/charts/blob/master/stable/mongodb-replicaset/templates/mongodb-poddisruptionbudget.yaml#L16

Comment (caarlos0) [reply stable/traefik/README.md]: PR updated :)

**Comment by caarlos0** [comment]
@krancour @unguiculus @timstoop changes made

**Comment by caarlos0** [comment]
> @krancour @unguiculus @timstoop

ping?

**Comment by caarlos0** [comment]
Thanks!

**Pull request merged**